openmc/mgxs/mgxs.py

```diff
@@ -675,14 +684,20 @@ def load_from_statepoint(self, statepoint):
             self.domain = statepoint.summary.get_universe_by_id(self.domain.id)
         elif self.domain_type == 'material':
             self.domain = statepoint.summary.get_material_by_id(self.domain.id)
+        elif self.domain_type == 'mesh':
+            self.domain = statepoint.meshes[self.domain.id]
         else:
             msg = 'Unable to load data from a statepoint for domain type {0} ' \
                   'which is not yet supported'.format(self.domain_type)
             raise ValueError(msg)
 
         # Use tally "slicing" to ensure that tallies correspond to our domain
         # NOTE: This is important if tally merging was used
-        if self.domain_type != 'distribcell':
+        if self.domain_type == 'mesh':
+            filters = [self.domain_type]
+            xyz = map(lambda x: np.arange(1, x+1), self.domain.dimension)
```

Comment:
The only thing PEP8 says is that you shouldn't assign a lambda. In this case, it's being used anonymously within a larger expression, so that would be fine per PEP8.